### PR TITLE
[Sarah + Smruthi] Completed like, dislike, delete a review. [Sarah] Completed update average star and price ratings in Review Summary section.

### PR DESCRIPTION
**Pull request by saraholiviawu**
**[Sarah + Smruthi] Description: like/dislike/delete a review**: A user can like or dislike any review. They can also do so on a review that they have posted themselves. The active thumbs or thumbs down is changed to the primary color and the number of likes / dislikes are updated accordingly. The user can delete the review that they have posted as well. There are six scenarios that are accounted for:
  - **User adds a like for the first time**: highlight thumbs up, increment number of likes
  - **User adds a dislike for the first time**: highlight thumbs down, increment number of dislikes
  - **User adds a like for the second time in a row (remove like)**: unhighlight thumbs up, decrement number of likes
  - **User adds a dislike for the second time in a row (remove dislike)**:  unhighlight thumbs down, decrement number of dislikes
  - **User who previously added a like now adds a dislike**: unhighlight thumbs up, decrement number of likes, highlight thumbs down, increment number of dislikes
   - **User who previously added a dislike now adds a like**: unhighlight thumbs down, decrement number of dislikes, highlight thumbs up, increment number of likes

**[Sarah] Description: Update average star and price ratings**: After a user posts a review, the average star and price are updated in the Review Summary section. A user can only have one review posted at all times, meaning that if the user were to add a review for the second time, their review ratings would be replaced by their latest ratings. When the user deletes a review, the average star and price ratings are adjusted accordingly. These two scenarios accounted for (which each have sub scenarios) as so:

  - **User adds a review for the first time**: Add star rating to cumulative sum of all star ratings users have left on this business previously. Increment number of star ratings. If there is a price rating (optional), add it to the cumulative sum of all price ratings left on this business. Increment number of price ratings. 
  - **User adds a review but has a preexisting review**: Get previous star rating and price rating. 
     -  **Star ratings**: Subtract old star rating from sum of all star ratings. Add new star rating to cumulative sum of all star ratings.
     - **Price ratings**:  
          - If user had a previously had no price rating and does not submit one in the newest review, do nothing. 
          - If a user previously had no price rating and submits one now, add to sum of all price ratings. Increment number of price ratings. 
          - If user previously had a price rating but does not submit a new one, subtract old price rating from sum of all price ratings and decrement number of price ratings.
          -  If the user had a price rating previously and also includes one now, subtract old price rating from sum of all price ratings and add the new price rating.

**Demo**:

1. Like / dislike a review 
See full video: https://drive.google.com/file/d/1TICirwwUeUfaHqfZ_j_rj9xjvNO0AdXO/view
BD Like _ Dislike a review

2. Add a review + update average star and price ratings 
See full video:https://drive.google.com/file/d/1WmhkhAZ0r8ohsNoD8-JPNpPSIH7SRVz-/view
BD Add a review (2)

3. Delete a review + update average star and price ratings
See full video: https://drive.google.com/file/d/1TICirwwUeUfaHqfZ_j_rj9xjvNO0AdXO/view
BD Delete a review

### CODE REVIEW

File: ripple/src/main/webapp/businessdetails.js [hunk 1]
```diff
@@ -275,14 +377,164 @@ function priceContent(priceRating) {
...
+  var i;
+  for (var i = 1; i <= 5; i++) {
+    elementId = "#star-" + i; 
+    $(elementId).removeClass("fa-star");
+    $(elementId).addClass("fa-star-o");
+    $(elementId).removeClass("star-color");
+  }
+}
+
+/* When a user deletes their review, hide dat a */
```
Comment (peter14f) [inline ripple/src/main/webapp/businessdetails.js:391]: What do you mean by "hide dat a"?

File: ripple/src/main/webapp/businessdetails.js [hunk 2]
```diff
@@ -88,65 +90,165 @@ function newReview() {
...
-        // Update list and push to Firestore
+      // Read summary ratings
+      var sumStarRatings = doc.data().sumStarRatings;
+      var numStarRatings = doc.data().numStarRatings;
+      var sumPriceRatings = doc.data().sumPriceRatings;
+      var numPriceRatings = doc.data().numPriceRatings;
+      // If user has not reviewed this business, add this user
+      console.log("updateUsersReviewedList() reviews summary: ", usersReviewedList, sumStarRatings, numStarRatings, sumPriceRatings, numPriceRatings);
+      if (!usersReviewedList.includes(uid)) {
+    //     // Update list and push to Firestore
```
Comment (peter14f) [inline ripple/src/main/webapp/businessdetails.js:120]: One "//" should be enough

File: ripple/src/main/webapp/businessdetails.js (hunk 1 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js]: Whoops, fixed

File: ripple/src/main/webapp/businessdetails.js (hunk 2 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js]: Fixed

**Pull request merged**